nexus/src/app/sagas/instance_start.rs

```diff
@@ -616,17 +622,37 @@ mod test {
         let _project_id = setup_test_project(&client).await;
         let opctx = instance_create::test::test_opctx(cptestctx);
         let instance = create_instance(client).await;
+
+        // Fetch enough state to be able to reason about how many nodes are in
+        // the saga.
         let db_instance =
             fetch_db_instance(cptestctx, &opctx, instance.identity.id).await;
-
         let params = Params {
             serialized_authn: authn::saga::Serialized::for_opctx(&opctx),
             instance: db_instance,
             ensure_network: true,
         };
-
         let dag = create_saga_dag::<SagaInstanceStart>(params).unwrap();
-        for node in dag.get_nodes() {
+        let num_nodes = dag.get_nodes().count();
```

Comment:
(an aside it'd be nice if `SagaDagIterator` implemented `ExactSizeIterator` and a more efficient `Iterator::nth`)

Reply:
Filed https://github.com/oxidecomputer/steno/issues/164